src/__tests__/components/WalletAddress.test.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { unmountComponentAtNode } from 'react-dom';
+import { render, act, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { WalletAddress } from '../../components/WalletAddress';
+import * as _ from 'lodash';
+import hathorLib from '@hathor/wallet-lib'
+
+const sampleAddress = 'WPhehTyNHTPz954CskfuSgLEfuKXbXeK3f';
+
+class ModalAddressQRCodeMock extends React.Component {
+  render() { return <div/> }
+}
+jest.mock('../../components/ModalAddressQRCode', () => () => {
+  const MockName = "modal-address-qrcode-mock";
+  return <MockName />;
+});
+
+// Mocking the underlying hathor wallet to better manage the tests
+let oldWallet;
+
+let container = null;
+beforeEach(() => {
+  // setup a DOM element as a render target
+  container = document.createElement("div");
+  document.body.appendChild(container);
+
+  // Mock the oldWallet object
+  oldWallet = hathorLib.wallet;
+  hathorLib.wallet = {
+    isHardwareWallet: () => false
+  }
+  hathorLib.wallet.isSoftwareWallet =
+    () => !hathorLib.wallet.isHardwareWallet()
```

Comment:
This one could have been as follows, to take advantage of the object literal:
```
  // Mock the oldWallet object
  const falsy = () => false;
  oldWallet = hathorLib.wallet;
  hathorLib.wallet = {
    isHardwareWallet: falsy,
    isSoftwareWallet: () => !falsy()
  }
```
I believe it would improve readability as well.

Reply:
This specific code would not work because we need to dynamically change the results on `isHardwareWallet` and `isSoftwareWallet`. Each is called on different places within the tested component, and they must return the same "answer" as to if the wallet is hardware or software.

The actual code of `isSoftwareWallet` is almost exactly this:
https://github.com/HathorNetwork/hathor-wallet-lib/blob/master/src/wallet.js#L2711-L2721

To improve legibility, I refactored it to be a "mini" class on on [58d6a76](https://github.com/HathorNetwork/hathor-wallet/pull/358/commits/58d6a763d77cd32852ab809f54fb09f00d4915fa). Do you think this makes it clearer what is the purpose of this mock?

Reply:
Interessante! Ficou bem mais legivel. Nice!